### DISCUSSION TITLE
Table Alias not getting right for softDelete on Model

### DISCUSSION THREAD

**Discussion by parmonov98** [laravel/framework#41708] [Q&A]
- Laravel Version:
- 
![image](https://user-images.githubusercontent.com/24564577/160352341-215790ff-44db-4b98-a16e-108d7952a01d.png)

- PHP Version: 8.0
- Database Driver & Version: MySql

### Description:
I enabled softDelete in Model using trait. and the tale has deleted_at column.
This query
EmployeeTransaction::query()->from("employee_transactions as et")
                    ->leftJoin('employees as e', 'e.id', '=', 'et.employee_id')
                    ->whereRaw("DATE_FORMAT(et.date, '%Y-%m-%d') >= '$this->from_date'")
                    ->whereRaw("DATE_FORMAT(et.date, '%Y-%m-%d') <= '$this->to_date'")
                    ->groupBy(DB::raw('et.employee_id'))
                    ->select(
                        DB::raw("`e`.`id` as employee_id"),
                        DB::raw("(SELECT SUM(`et`.`earning_amount`) - SUM(`et`.`payment_amount`) from employee_transactions as et where et.employee_id = e.id GROUP BY et.employee_id) as left_amount"),
                        DB::raw("CONCAT(`e`.`first_name`, '', `e`.`last_name`) as name"),
                        DB::raw("SUM(`et`.`earning_amount`) as earning_amount"),
                        DB::raw("SUM(`et`.`payment_amount`) as payment_amount"),
                        DB::raw("SUM(earning_amount - payment_amount) as difference_amount"),
                        DB::raw("`et`.`date` as date")
                    )

It throws an error: 
![image](https://user-images.githubusercontent.com/24564577/160353000-86fa4761-82a7-4bec-83e4-0f6acff059cb.png)

### Steps To Reproduce:
create a eloquent query using model and change table name using from or fromRaw

**Comment by gguney** [comment]
Can you also share employee_transactions table columns? 

Did you add softDeletes() to your migration like:
```

        Schema::create('employee_transactions', function (Blueprint $table) {
            $table->id();
            ...
            $table->softDeletes();
        });
```

**Reply by parmonov98** [reply]
![image](https://user-images.githubusercontent.com/24564577/163836328-4dfde949-6298-4401-8b44-d4e38a6ac6a3.png)

**Comment by krkarel** [comment]
Still an issue in 2024...
`"laravel/framework": "^10.9"`

**Reply by cgarciagarcia** [reply]
yeah, could you watch the new post here https://github.com/laravel/framework/pull/52294

**Comment by macropay-solutions** [comment]
Raw queries should be avoided when possible.

Please try this:

```
EmployeeTransaction::query()->setModel(($et = new EmployeeTransaction())->setTable($et->getTable() . ' as et'))
->leftJoin('employees as e', 'e.id', '=', 'et.employee_id')
->whereRaw("DATE_FORMAT(et.date, '%Y-%m-%d') >= '$this->from_date'")
->whereRaw("DATE_FORMAT(et.date, '%Y-%m-%d') <= '$this->to_date'")
->groupBy(DB::raw('et.employee_id'))
->select(
DB::raw("e.id as employee_id"),
DB::raw("(SELECT SUM(et.earning_amount) - SUM(et.payment_amount) from employee_transactions as et where et.employee_id = e.id GROUP BY et.employee_id) as left_amount"),
DB::raw("CONCAT(e.first_name, '', e.last_name) as name"),
DB::raw("SUM(et.earning_amount) as earning_amount"),
DB::raw("SUM(et.payment_amount) as payment_amount"),
DB::raw("SUM(earning_amount - payment_amount) as difference_amount"),
DB::raw("et.date as date")
)
```

**Comment by cgarciagarcia** [comment]
I wrote a solution fot this. here you are the [PR](https://github.com/laravel/framework/pull/52294), is a rudimentary way to resolve this problem. Let's see what happen.
